### PR TITLE
feat: add duplicate setting name detection and conflict handling

### PR DESCRIPTION
**Pull request by grigoriev**
### Proposed changes

- Introduce `DuplicateSettingNameException` to handle setting name conflicts.
- Add `DuplicateSettingNameExceptionMapper` for REST API exception mapping.
- Implement duplicate name detection in `GenericNamedSettings`.
- Add unit tests for duplicate name handling logic.

### Checklist

Before creating a PR, run through this checklist and mark each as complete:
- [x] I have read the [`CONTRIBUTING`](CONTRIBUTING.md) document
- [x] If applicable, I have added tests that prove my fix is effective or that my feature works
- [x] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] I have updated any relevant documentation

### CODE REVIEW

File: app/src/test/java/ch/sbb/polarion/extension/generic/rest/exception/mapper/DuplicateSettingNameExceptionMapperTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    @Test
+    void testResponse() {
+        try (Response response = new DuplicateSettingNameExceptionMapper().toResponse(new DuplicateSettingNameException("test message"))) {
+            assertEquals(Response.Status.CONFLICT.getStatusCode(), response.getStatus());
+            ErrorEntity entity = (ErrorEntity) response.getEntity();
+            assertNotNull(entity);
+            assertEquals("test message", entity.getMessage());
+        } catch (Exception e) {
+            fail("Exception thrown: " + e.getMessage());
```
[the remark this answers is not in the document]
Comment (grigoriev) [reply app/src/test/java/ch/sbb/polarion/extension/generic/rest/exception/mapper/DuplicateSettingNameExceptionMapperTest.java]: Removed

**Review by yurtsevich-sbb — APPROVED** [review]
